### PR TITLE
MEN-4268: Added unit tests for config.py

### PR DESCRIPTION
**Pull request by NilsOlavKJohansen**
Changelog: None
Signed-off-by: nils olav <nils.olav@northern.tech>

### CODE REVIEW

File: tests/unit/test_config.py [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+@pytest.fixture(scope="session", name="local_and_global")
+def fixture_local_and_global():
+    return config.load(
+        "tests/unit/data/configs/local_mender.conf",
+        "tests/unit/data/configs/global_medner.conf",
+    )
+
+
+@pytest.fixture(scope="session", name="glob_only")
+def fixture_glob_only():
```
Comment (oleorhagen) [inline tests/unit/test_config.py:55]: `s/glob/global/`

File: tests/unit/test_config.py [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+    def test_local_priority(self, both_with_local_priority):
+        # Local IventoryPollIntervalSeconds == 100 and Global IventoryPollIntervalSeconds == 200
+        assert both_with_local_priority.InventoryPollIntervalSeconds == 100
+
+    def test_with_no_local_server_url(self, both_with_local_priority):
+        # Local serveURL is non existing and Global is https://hosted.mender.io
+        assert both_with_local_priority.ServerURL == "https://hosted.mender.io"
+
+
+class TestFaultyJsonfile:
```
Comment (oleorhagen) [inline tests/unit/test_config.py:151]: `s/Json/JSON/`

File: tests/unit/test_config.py [hunk 3]
```diff
@@ -0,0 +1,209 @@
...
+def fixture_local_only():
+    with open("tests/unit/data/configs/local_mender_testdata.conf", "w") as f:
+        json.dump(LOCAL_TESTDATA, f)
+    yield config.load("", "tests/unit/data/configs/local_mender_testdata.conf")
+    if os.path.isfile("tests/unit/data/configs/local_mender_testdata.conf"):
+        os.remove("tests/unit/data/configs/local_mender_testdata.conf")
+
+
+@pytest.fixture(scope="session", name="both_with_local_priority")
+def fixture_both_with_local_priority():
```
Comment (oleorhagen) [inline tests/unit/test_config.py:73]: What does this mean?
Can they ever both have the same priority? :thinking:

Comment (NilsOlavKJohansen) [reply tests/unit/test_config.py:73]: No, that means that information is in both local and global, but the local has priority. Poor name, I will change it:)

File: tests/unit/test_config.py [hunk 4]
```diff
@@ -0,0 +1,209 @@
...
+        assert global_only.ServerCertificate == ""
+
+
+class TestLocalPriority:
+    def test_local_priority(self, local_priority):
+        # Local IventoryPollIntervalSeconds == 100 and Global IventoryPollIntervalSeconds == 200
+        assert local_priority.InventoryPollIntervalSeconds == 100
+
+    def test_with_no_local_server_url(self, local_priority):
+        # Local serveURL is non existing and Global is https://hosted.mender.io
```
Comment (oleorhagen) [inline tests/unit/test_config.py:147]: `s/serveURL/serverURL/`

**Comment by oleorhagen** [comment]
@NilsOlavKJohansen one more spelling error

**Review by oleorhagen — APPROVED** [review]

**Pull request merged**